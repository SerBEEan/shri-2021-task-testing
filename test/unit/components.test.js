import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import { ProductItem } from '../../src/client/components/ProductItem';
import { ProductDetails } from '../../src/client/components/ProductDetails';
import { Cart } from '../../src/client/pages/Cart';

const PRODUCT = { id: 1, name: 'product 1', price: 1000, color: 'red', description: 'description', material: 'brick' };
const PRODUCT_SHORT_INFO = { id: 1, name: 'product 1', price: 1000 };

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: () => ({
        [1]: { name: 'product 1', price: 1000, count: 1 },
        [2]: { name: 'product 2', price: 800, count: 2 },
        [3]: { name: 'product 3', price: 1500, count: 1 },
    }),
}));


describe('компоненты', () => {
    it('снимок карточки товара', () => {
        const wrapper = shallow(<ProductItem product={PRODUCT_SHORT_INFO} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('снимок страницы с деталями товара', () => {
        const wrapper = shallow(<ProductDetails product={PRODUCT} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('снимок корзины с добавленными товарами', () => {
        const wrapper = shallow(<Cart />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('считается общая сумма товаров в корзине', () => {
        const wrapper = shallow(<Cart />);

        const allPrice = wrapper.find('.Cart-OrderPrice').text();

        expect(allPrice).toEqual('$4100');
    });
});
