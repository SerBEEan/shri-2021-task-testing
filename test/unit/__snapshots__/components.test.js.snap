// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`компоненты снимок карточки товара 1`] = `
<div
  className="ProductItem card w-100 mb-4"
  data-testid={1}
>
  <Image
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <h5
      className="ProductItem-Name card-title"
    >
      product 1
    </h5>
    <p
      className="ProductItem-Price card-text"
    >
      $
      1000
    </p>
    <Link
      className="ProductItem-DetailsLink card-link"
      to="/catalog/1"
    >
      Details
    </Link>
    <CartBadge
      id={1}
    />
  </div>
</div>
`;

exports[`компоненты снимок корзины с добавленными товарами 1`] = `
<div
  className="Cart"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Shopping cart"
  />
  <div
    className="row mb-4"
  >
    <div
      className="col"
    >
      <h1>
        Shopping cart
      </h1>
      <table
        className="Cart-Table table"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              Product
            </th>
            <th
              scope="col"
            >
              Price
            </th>
            <th
              scope="col"
            >
              Count
            </th>
            <th
              scope="col"
            >
              Total
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="1"
            key="1"
          >
            <th
              className="Cart-Index"
              scope="row"
            >
              1
            </th>
            <td
              className="Cart-Name"
            >
              product 1
            </td>
            <td
              className="Cart-Price"
            >
              $
              1000
            </td>
            <td
              className="Cart-Count"
            >
              1
            </td>
            <td
              className="Cart-Total"
            >
              $
              1000
            </td>
          </tr>
          <tr
            data-testid="2"
            key="2"
          >
            <th
              className="Cart-Index"
              scope="row"
            >
              2
            </th>
            <td
              className="Cart-Name"
            >
              product 2
            </td>
            <td
              className="Cart-Price"
            >
              $
              800
            </td>
            <td
              className="Cart-Count"
            >
              2
            </td>
            <td
              className="Cart-Total"
            >
              $
              1600
            </td>
          </tr>
          <tr
            data-testid="3"
            key="3"
          >
            <th
              className="Cart-Index"
              scope="row"
            >
              3
            </th>
            <td
              className="Cart-Name"
            >
              product 3
            </td>
            <td
              className="Cart-Price"
            >
              $
              1500
            </td>
            <td
              className="Cart-Count"
            >
              1
            </td>
            <td
              className="Cart-Total"
            >
              $
              1500
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colSpan={4}
            >
              Order price:
            </td>
            <td
              className="Cart-OrderPrice"
            >
              $
              4100
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-6"
    >
      <button
        className="Cart-Clear btn btn-outline-secondary"
        onClick={[Function]}
      >
        Clear shopping cart
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12 col-sm-8 col-md-6 col-lg-4"
    >
      <h2>
        Сheckout
      </h2>
      <Form
        onSubmit={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`компоненты снимок страницы с деталями товара 1`] = `
<div
  className="ProductDetails row"
>
  <div
    className="col-12 col-sm-5 col-lg-4"
  >
    <Image />
  </div>
  <div
    className="col-12 col-sm-7 col-lg-6"
  >
    <h1
      className="ProductDetails-Name"
    >
      product 1
    </h1>
    <p
      className="ProductDetails-Description"
    >
      description
    </p>
    <p
      className="ProductDetails-Price fs-3"
    >
      $
      1000
    </p>
    <p>
      <button
        className="ProductDetails-AddToCart btn btn-primary btn-lg"
        onClick={[Function]}
      >
        Add to Cart
      </button>
      <CartBadge
        id={1}
      />
    </p>
    <dl>
      <dt>
        Color
      </dt>
      <dd
        className="ProductDetails-Color text-capitalize"
      >
        red
      </dd>
      <dt>
        Material
      </dt>
      <dd
        className="ProductDetails-Material text-capitalize"
      >
        brick
      </dd>
    </dl>
  </div>
</div>
`;
